pkg/cloud/azure/actuators/cluster/reconciler.go

```diff
@@ -151,6 +179,52 @@ func (s *Reconciler) Reconcile() error {
 		return errors.Wrapf(err, "failed to createorupdate control plane public load balancer for cluster %s", s.scope.Cluster.Name)
 	}
 
+	actuators.CreateOrUpdateBastionIP(s.scope)
+
+	bastionPublicIPSpec := &publicips.Spec{
+		Name: s.scope.Network().BastionIP.Name,
+	}
+	if err := s.publicIPSvc.CreateOrUpdate(s.scope.Context, bastionPublicIPSpec); err != nil {
+		return errors.Wrapf(err, "failed to createorupdate bastion public ip for cluster %s", s.scope.Cluster.Name)
+	}
+
+	bastionNicSpec := &networkinterfaces.Spec{
```

Comment:
can this be part of machine actuator itself, so we can define 3 different roles, node, controlplane and bastion. so we can specify this as part of cluster spec itself instead of hard coding values here